includes/tracking-events.php

```diff
@@ -0,0 +1,120 @@
+<?php
+    if(class_exists('WooCommerce')){
+        // /** check certain tracking scenarios **/
+        // if visitor is viewing product
+        if(!$this->single_item_tracked && is_product()){
+            $product = $this->resolve_product(get_queried_object_id());
+            $product_id = method_exists($product, 'get_id') ? $product->get_id() : $product->id;
+            $this->put_event_in_queue(
+                "window.metrilo.viewProduct('"
+                . $product_id
+                . "');"
+            );
+            $this->single_item_tracked = true;
+        }
+        
+        // if visitor is viewing product category
+        if (!$this->single_item_tracked && is_product_category()) {
+            $category = get_queried_object();
+            $this->put_event_in_queue(
+                "window.metrilo.viewCategory('"
+                . $category->term_id
+                . "');"
+                
+            );
+            $this->single_item_tracked = true;
+        }
+    
+        // if visitor is viewing search page
+        if (!$this->single_item_tracked && is_search()) {
+            $search = get_queried_object();
+            $this->put_event_in_queue(
+                "window.metrilo.search('"
+                . get_search_query()
+                . "', '"
+                . get_search_link()
+                . "');"
```

Comment:
Isn't this a bit more readable?
```suggestion
                $query = get_search_query();
                $link = get_search_link();
                
                "window.metrilo.search('$query', '$link');"
```
It applies to other similar places as well.

Also, what if the search query string has any quotes or injection strings?

Reply:
Did some code refactoring about this, changed all of the above format to your suggestion.